src/Compilers/Core/Portable/Operations/BasicBlock.cs

```diff
@@ -24,6 +27,8 @@ public sealed class BasicBlock
     {
         private readonly ImmutableArray<IOperation>.Builder _statements;
         private readonly ImmutableHashSet<BasicBlock>.Builder _predecessors;
+        internal Branch InternalNext;
+        internal (IOperation Condition, bool JumpIfTrue, Branch Branch) InternalConditional;
```

Comment:
Why move these to internal fields and not keep the properties? Seems to me that will just make it harder if we ever want to add assertions about any of these values.

Reply:
> Why move these to internal fields and not keep the properties? 

These are structures and they cannot be modified through properties. Individual members of these structures are modified independently. 

> Seems to me that will just make it harder if we ever want to add assertions about any of these values.

Well, we will cross this bridge when we get to it, if ever.

---
In reply to: [167649176](https://github.com/dotnet/roslyn/pull/24770#discussion_r167649176) [](ancestors = 167649176)